src/runtime/tuple_parser.c

```diff
@@ -52,49 +49,59 @@ static parser combinate(heap h, selfparser p)
     return (parser)closure(h, selfinate, h, p);
 }
 
-
 closure_function(1, 2, parser, til_newline,
                  parser, finish,
                  parser, self, character, in)
 {
-    if (in == '\n') return bound(finish);
+    if (in == '\n') {
+        parser f = bound(finish);
+        deallocate_closure(self);
```

Comment:
I thought closure_finish does deallocate the closure (?)

Reply:
Right, but this was deallocating the closure from selfinate - but I think this can go away.